src/text/span.rs

```diff
@@ -155,3 +161,73 @@ impl<'a> Styled for Span<'a> {
         self
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn from_ref_str_borrowed_cow() {
+        let content = "some string";
+
+        let span = Span::from(content);
+
+        assert_eq!(span.content, Cow::Borrowed(content));
+    }
+
+    #[test]
+    fn from_string_ref_str_borrowed_cow() {
+        let content = String::from("some string");
+
+        let span = Span::from(content.as_str());
+
+        assert_eq!(span.content, Cow::Borrowed(content.as_str()));
+    }
+
+    #[test]
+    fn from_string_owned_cow() {
+        let content = String::from("some string");
+        let content_clone = content.clone();
+
+        let span = Span::from(content);
+
+        assert_eq!(span.content, Cow::Owned::<str>(content_clone));
+    }
+
+    #[test]
+    fn from_ref_string_borrowed_cow() {
+        let content = String::from("some string");
+
+        let span = Span::from(&content);
+
+        assert_eq!(span.content, Cow::Borrowed(content.as_str()));
+    }
```

Comment:
Nit: I understand the AAA approach, but for simple tests dropping the spaces make the tests easier to compare without  scrolling. A couple of small rearrangements too:

```suggestion
    #[test]
    fn from_ref_str_borrowed_cow() {
        let content = "some string";
        let span = Span::from(content);
        assert_eq!(span.content, Cow::Borrowed(content));
    }

    #[test]
    fn from_ref_string_borrowed_cow() {
        let content = String::from("some string");
        let span = Span::from(&content);
        assert_eq!(span.content, Cow::Borrowed(content.as_str()));
    }

    #[test]
    fn from_string_ref_str_borrowed_cow() {
        let content = String::from("some string");
        let span = Span::from(content.as_str());
        assert_eq!(span.content, Cow::Borrowed(content.as_str()));
    }

    #[test]
    fn from_string_owned_cow() {
        let content = String::from("some string");
        let span = Span::from(content.clone());
        assert_eq!(span.content, Cow::Owned::<str>(content));
    }
```